app/models/page/DashboardPage.scala

```diff
@@ -16,111 +16,251 @@
 
 package models.page
 
-import models.ApplicationData.ApplicationStatus
-import models.ApplicationData.ApplicationStatus.ApplicationStatus
-import models.page.DashboardPage.ProgressStepVisibility
-import models.{ ApplicationData, CachedData, Progress }
+import connectors.{ AllocationExchangeObjects, ExchangeObjects }
+import models.page.DashboardPage.Flags.{ ProgressInactive, _ }
+import models.{ CachedData, Progress }
+import org.joda.time.LocalDate
 import play.api.i18n.Lang
 import play.api.mvc.RequestHeader
-import security.RoleUtils
-import security.Roles.DisplayOnlineTestSectionRole
+import security.RoleUtils._
+import security.Roles._
 
-// format: OFF
 case class DashboardPage(firstStepVisibility: ProgressStepVisibility,
                          secondStepVisibility: ProgressStepVisibility,
                          thirdStepVisibility: ProgressStepVisibility,
-                         fourthStepVisibility: ProgressStepVisibility)
-
-// format: ON
+                         fourthStepVisibility: ProgressStepVisibility,
+                         isApplicationSubmittedAndNotWithdrawn: Boolean,
```

Comment:
I like the idea of these flags to avoid adding logic to the view. However, once we implement Withdraw stories we may need to rethink this because in theory and from the point of view of application status you can only be SUBMITTED or WITHDRAWN. Do we really want to distinguish between if you have or haven't submitted? We did in fasttrack but in faststream is it necessary? The same applies with some of the other flags.

Generally speaking it seems there are many flags/ conditions maybe we can/ should simplify the logic, we may need to involve BA for that. Also the flags have names that basically are the logic itself but in a single flag/name (isApplicationSubmittedAndNotWithdrawn). Why do not we follow a similar approach is? isFirstStepVisible? isSecondStepVisible? cannot we say isWithdrawButtonVisible, isFirstSectionEnable, etc?